kasa/protocol.py

```diff
@@ -32,6 +33,54 @@
 _NO_RETRY_ERRORS = {errno.EHOSTDOWN, errno.EHOSTUNREACH, errno.ECONNREFUSED}
 
 
+def md5(payload: bytes) -> bytes:
+    """Return an md5 hash of the payload."""
+    digest = hashes.Hash(hashes.MD5())  # noqa: S303
+    digest.update(payload)
+    hash = digest.finalize()
+    return hash
+
+
+class TPLinkTransport(ABC):
```

Comment:
```suggestion
class Transport(ABC):
```
or `TransportProtocol`? I'd rather like to see us getting rid off superfluous prefixes if possible, or at least try to not add more of them.

Reply:
Done, renamed BaseTransport